app/javascript/mastodon/features/notifications/components/setting_toggle.js

```diff
@@ -3,12 +3,28 @@ import PropTypes from 'prop-types';
 import ImmutablePropTypes from 'react-immutable-proptypes';
 import Toggle from 'react-toggle';
 
-const SettingToggle = ({ settings, settingKey, label, onChange, htmlFor = '' }) => (
-  <label htmlFor={htmlFor} className='setting-toggle__label'>
-    <Toggle checked={settings.getIn(settingKey)} onChange={(e) => onChange(settingKey, e.target.checked)} />
-    <span className='setting-toggle'>{label}</span>
-  </label>
-);
+class SettingToggle extends React.PureComponent {
```

Comment:
On the first page load (`/web/getting-started` on desktop) is renders 14 times. Did not investigate further.